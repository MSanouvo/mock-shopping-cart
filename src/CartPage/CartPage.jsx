import NavBar from "../components/Navbar/NavBar";
import CartItem from "../components/CartItems/CartItem";
import PropTypes from "prop-types";
import { useState } from "react";
import styles from "./CartPage.module.css";
import Button from "../components/button/button";

CartPage.propTypes = {
  cart: PropTypes.array,
  total: PropTypes.number
}

CartPage.defaultProps = {
  cart: [],
};

function CartPage(props) {
  const [cart, setCart] = useState(props.cart);

  const deleteItem = (item) => {
    const newCart = [...cart];
    for (let i = 0; i < newCart.length; i++) {
      if (newCart[i].name === item) {
        newCart.splice(i, 1);
        setCart(newCart);
      }
    }
    console.log(cart);
  };
  return (
    <div className={styles.container}>
      {/* <NavBar total={props.total} /> */}
      <main className={styles.cart}>
        <header className={styles.heading}>Shopping Cart</header>
        {cart.length != 0 ? (
          cart.map((item, index) => {
            return (
              <div key={index} className={styles.cartItems}>
                <CartItem
                  key={index}
                  name={item.name}
                  count={item.count}
                  src={item.src}
                  deleteItem={deleteItem}
                />
              </div>
            );
          })
        ) : (
          <p className={styles.message}>Add items to cart</p>
        )}
      </main>
      <div className={styles.button}>
        {cart.length != 0 && <Button name={"Complete Transaction"} />}
      </div>
    </div>
  );
}

export default CartPage;
