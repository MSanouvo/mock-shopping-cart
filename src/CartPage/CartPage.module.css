.container{
    /* border: 1px solid yellow; */
    min-width: 100vw;
}

.heading{
    font-weight: bold;
    font-size: 2.5rem;
    padding: 20px;
    color: black;
}

.cart{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: start;
    gap: 20px;
    padding: 10px 20px;
    color: black;
    /* outline: 1px solid black; */
}

.cartItems{
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    gap: 30px;
    box-shadow: 0px 0px 5px black;
}

.button{
    display: flex;
    /* border: 1px solid green; */
    padding: 30px;
    width: 95%;
    justify-content: end;
}

.message{
    width: 100%;
    text-align: center;
}