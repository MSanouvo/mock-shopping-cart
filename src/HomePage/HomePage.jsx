import Banner from "../assets/clothes-line.jpg";
import styles from "./HomePage.module.css";

function HomePage() {
  return (
    <div className={styles.container}>
      <div className={styles.content}>
        <img className={styles.banner} src={Banner} alt="store banner" />
        <main className={styles.message}>
          <p>Welcome to my mock digital storefront!</p>
          <p className={styles.text}>
            Here you can shop for items by adding them to your cart and
            completing your transaction. You can begin by pressing on shop in
            the top right corner to see the store page. Once items have been
            added to the cart you can click on the cart icon to see them
            displayed there. If you change your mind about an item you can
            always delete it from the cart befor completing your transaction.
          </p>
        </main>
      </div>
    </div>
  );
}

export default HomePage;
