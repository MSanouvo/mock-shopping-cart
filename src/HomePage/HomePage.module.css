.banner {
  height: 50vh;
}

.container {
  /* border: 1px solid yellow; */
  min-width: 100vw;
}

.content {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: 10px;
  padding: 30px 50px;
}

.message {
  font-size: 2rem;
  color: black;
  text-align: center;
}

.text {
  font-size: 1.2rem;
  width: 70vw;
}
