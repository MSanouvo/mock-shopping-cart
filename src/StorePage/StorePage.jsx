import StoreItem from "../components/StoreItems/StoreItem";
import PropTypes from "prop-types";
import { useState, useEffect } from "react";
import styles from "./StorePage.module.css";

StorePage.propTypes = {
  callItem: PropTypes.func
}

function StorePage(props) {
  const [inventory, setInventory] = useState(null);

  useEffect(() => {
    fetch("https://fakestoreapi.com/products/")
      .then((res) => res.json())
      .then((res) => setInventory(res.splice(0, 12)));
  }, []);

  return (
    <div className={styles.container}>
      <main className={styles.store}>
        <header className={styles.heading}>Store</header>
        <div className={styles.items}>
          {inventory != null ? (
            inventory.map((item, index) => {
              return (
                <div key={index}>
                  <StoreItem
                    key={index}
                    name={item.title}
                    count={1}
                    callItem={props.callItem}
                    src={item.image}
                    alt={item.title}
                  />
                </div>
              );
            })
          ) : (
            <p className={styles.message}>Loading items...</p>
          )}
        </div>
      </main>
    </div>
  );
}

export default StorePage;
