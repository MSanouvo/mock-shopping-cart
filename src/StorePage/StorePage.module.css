.container{
    /* border: 1px solid yellow; */
    display: flex;
    flex-direction: column;
    height: 100vh;
    min-width: 100vw;
    overflow-y: scroll;
}

.heading{
    font-weight: bold;
    font-size: 2.5rem;
    padding: 20px;
    color: black;
}

.store{
    display: flex;
    flex-direction: column;
    padding: 10px 20px;
}

.items{
    /* outline: 1px solid purple; */
    display: grid;
    grid-template-rows: 100px, 1fr;
    grid-template-columns: repeat(3, 30%);
    justify-content: center;
    height: 500px;
    gap: 30px;
    grid-row: 2;
    padding: 10px;
}

.message{
    width: 100%;
    text-align: center;
}