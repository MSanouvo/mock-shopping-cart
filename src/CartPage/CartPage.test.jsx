import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Route, Routes } from "react-router-dom";
import { MemoryRouter } from "react-router-dom";
import CartPage from "./CartPage";
import HomePage from "../HomePage/HomePage";
import StorePage from "../StorePage/StorePage";

describe("Cart page", () => {
  it("renders the Cart Page", () => {
    const { container } = render(
      <MemoryRouter>
        <CartPage />
      </MemoryRouter>,
    );
    expect(container).toMatchSnapshot();
  });

  it("renders the heading components", () => {
    render(
      <MemoryRouter>
        <CartPage />
      </MemoryRouter>,
    );

    expect(screen.getByText("Shopping Cart")).toBeInTheDocument();
  });

  it("renders message when cart is empty", () => {
    render(
      <MemoryRouter>
        <CartPage />
      </MemoryRouter>,
    );
    expect(screen.getByText("Add items to cart"));
  });

  it("renders item when cart is not empty", () => {
    render(
      <MemoryRouter>
        <CartPage cart={[{ name: "item", count: 1 }]} />
      </MemoryRouter>,
    );
    expect(screen.getByRole("heading", { name: "item" })).toBeInTheDocument();
    expect(screen.getByDisplayValue("1")).toBeInTheDocument();
  });

  it("deletes item on clicking delete button", async () => {
    const user = userEvent.setup();
    render(
      <MemoryRouter>
        <CartPage cart={[{ name: "item", count: 1 }]} />
      </MemoryRouter>,
    );

    const link = screen.getByText("Delete");
    await user.click(link);
    expect(screen.getByText("Add items to cart"));
  });


  //Obselete Tests after removing navbar from component
  // it("navigates to home page", async () => {
  //   render(
  //     <MemoryRouter initialEntries={["/cart"]}>
  //       <Routes>
  //         <Route path="/cart" element={<CartPage />} />
  //         <Route path="/" element={<HomePage />} />
  //       </Routes>
  //     </MemoryRouter>,
  //   );

  //   const user = userEvent.setup();
  //   const homeLink = screen.getByRole("link", { name: "Home" });

  //   await user.click(homeLink);
  //   expect(
  //     screen.getByText("Welcome to my mock digital storefront!"),
  //   ).toBeInTheDocument();
  // });

  // it("navigates to store page", async () => {
  //   render(
  //     <MemoryRouter initialEntries={["/cart"]}>
  //       <Routes>
  //         <Route path="/store" element={<StorePage />} />
  //         <Route path="/cart" element={<CartPage />} />
  //       </Routes>
  //     </MemoryRouter>,
  //   );

  //   const user = userEvent.setup();
  //   const link = screen.getByRole("link", { name: "Shop" });

  //   await user.click(link);
  //   expect(screen.getByText("Store")).toBeInTheDocument();
  // });
});
