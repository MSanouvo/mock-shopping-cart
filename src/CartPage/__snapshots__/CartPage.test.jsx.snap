// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cart page > renders the Cart Page 1`] = `
<div>
  <div
    class="_container_bc0b15"
  >
    <main
      class="_cart_bc0b15"
    >
      <header
        class="_heading_bc0b15"
      >
        Shopping Cart
      </header>
      <p
        class="_message_bc0b15"
      >
        Add items to cart
      </p>
    </main>
    <div
      class="_button_bc0b15"
    />
  </div>
</div>
`;
